cpp/arcticdb/version/version_map_batch_methods.hpp

```diff
@@ -117,6 +120,130 @@ inline std::shared_ptr<std::unordered_map<std::pair<StreamId, VersionId>, AtomKe
     return output;
 }
 
+struct StreamVersionData {
+    explicit StreamVersionData(const pipelines::VersionQuery& version_query) {
+        react(version_query);
+    }
+
+    StreamVersionData() = default;
+
+    size_t count_ = 0;
+    LoadParameter load_param_ = LoadParameter{LoadType::LOAD_LATEST_UNDELETED};
+
+    void react(const pipelines::VersionQuery& version_query) {
+        util::variant_match(version_query.content_, [that = this](const auto &query) {
+            that->do_react(query);
+        });
+    }
+
+    void do_react(std::monostate) {
+        ++count_;
+    }
+
+    void do_react(const pipelines::SpecificVersionQuery& specific_version) {
+        ++count_;
+        switch(load_param_.load_type_) {
+        case LoadType::LOAD_LATEST_UNDELETED:
+            load_param_ = LoadParameter{LoadType::LOAD_DOWNTO, specific_version.version_id_};
+            break;
+        case LoadType::LOAD_DOWNTO:
+            util::check(load_param_.load_until_.has_value(), "Expect LOAD_DOWNTO to have version specificed");
+            load_param_.load_until_ = std::min(load_param_.load_until_.value(), specific_version.version_id_);
+            break;
+        case LoadType::LOAD_FROM_TIME:
```

Comment:
I like annotating these cases with the [fallthrough](https://en.cppreference.com/w/cpp/language/attributes/fallthrough) attribute

Reply:
In this particular case, we are adding a break; thought...but could be a good practice